### PR TITLE
v0.7 compatability

### PR DESCRIPTION
**Pull request by ZacLN**
1. emits dotted operators (e.g. `a .+ b`) as single tokens
2. handles suffixed operators
3. adds ⟂
4. adds explicit BIN/HEX/OCT_INTs

### CODE REVIEW

**Comment by KristofferC** [comment]
It is great to see I am not the only one having a problem writing the word "compatability".

File: src/utilities.jl [hunk 1]
```diff
@@ -198,3 +198,184 @@ eof(c::Char) = c === EOF_CHAR
 
 readchar(io::IO) = eof(io) ? EOF_CHAR : read(io, Char)
 takechar(io::IO) = (readchar(io); io)
+
+# Checks whether a Char is an operator, which can not be juxtaposed with another
+# Char to be an operator (i.e <=), and can be prefixed by a dot (.)
```
Comment (KristofferC) [inline src/utilities.jl:203]: Can we get some comment here where all these "magic numbers" come from?

File: src/utilities.jl [hunk 2]
```diff
@@ -198,3 +198,184 @@ eof(c::Char) = c === EOF_CHAR
...
+    Tokens.NOT_SIGN <= k <= Tokens.QUAD_ROOT
+    ) 
+end
+
+# suffix operators
+# "₀₁₂₃₄₅₆₇₈₉₊₋₌₍₎²³¹ʰʲʳʷʸˡˢˣᴬᴮᴰᴱᴳᴴᴵᴶᴷᴸᴹᴺᴼᴾᴿᵀᵁᵂᵃᵇᵈᵉᵍᵏᵐᵒᵖᵗᵘᵛᵝᵞᵟᵠᵡᵢᵣᵤᵥᵦᵧᵨᵩᵪᶜᶠᶥᶦᶫᶰᶸᶻᶿ ⁰ⁱ⁴⁵⁶⁷⁸⁹⁺⁻⁼⁽⁾ⁿₐₑₒₓₕₖₗₘₙₚₛₜⱼⱽ′″‴‵‶‷⁗"
+
+
+# Used to generate dotop1 and isopsuffix
+function _genchartest(x::Vector{UInt32}, fname)
```
Comment (KristofferC) [inline src/utilities.jl]: Oh, they are generated from here? What is the reason to write the function out manually?

Comment (ZacLN) [reply src/utilities.jl]: The input vectors are created by filtering the operator list according to how they parse ( I've forgotten to include the code), which we don't want to do on startup

Comment (KristofferC) [reply src/utilities.jl]: But this would only happen during the precompilation phase?

Comment (ZacLN) [reply src/utilities.jl]: I thinks it's slightly more transparent this way and some of it was done by hand

Comment (Keno) [reply src/utilities.jl]: I also don't like having both the manual and the generated version in here. We should either have one or the other.

Comment (ZacLN) [reply src/utilities.jl]: fair enough, stored [here](https://gist.github.com/ZacLN/092430d37c5b31479b5c7bbb1d93bf82) for ref

**Comment by ZacLN** [comment]
Good to go? + tag as v0.5.0, or wait for rc1?

**Comment by KristofferC** [comment]
Sounds good to me, might be worth trying the benchmark just to see that nothing bad crept in.

**Comment by ZacLN** [comment]
I've just seen `.=>` has been added to master, will amend here

**Comment by ZacLN** [comment]
Good call on benchmarking- caught one

**Pull request merged**